core/src/main/java/io/grpc/util/OutlierDetectionLoadBalancer.java

```diff
@@ -0,0 +1,1001 @@
+          checkArgument(requestVolume != null);
+          this.requestVolume = requestVolume;
+          return this;
+        }
+
+        /** Builds a new instance of {@link SuccessRateEjection}. */
+        public SuccessRateEjection build() {
+          return new SuccessRateEjection(stdevFactor, enforcementPercentage, minimumHosts,
+              requestVolume);
+        }
+      }
+    }
+
+    /** The configuration for failure percentage ejection. */
+    public static class FailurePercentageEjection {
+      final Integer threshold;
+      final Integer enforcementPercentage;
+      final Integer minimumHosts;
+      final Integer requestVolume;
+
+      FailurePercentageEjection(Integer threshold, Integer enforcementPercentage,
+          Integer minimumHosts, Integer requestVolume) {
+        this.threshold = threshold;
+        this.enforcementPercentage = enforcementPercentage;
+        this.minimumHosts = minimumHosts;
+        this.requestVolume = requestVolume;
+      }
+
+      /** For building new {@link FailurePercentageEjection} instances. */
+      public static class Builder {
+        Integer threshold = 85;
+        Integer enforcementPercentage = 100;
+        Integer minimumHosts = 5;
+        Integer requestVolume = 50;
+
+        /** The failure percentage that will result in an address being considered an outlier. */
+        public Builder setThreshold(Integer threshold) {
+          checkArgument(threshold != null);
+          this.threshold = threshold;
+          return this;
+        }
+
+        /** Only eject this percentage of outliers. */
+        public Builder setEnforcementPercentage(Integer enforcementPercentage) {
+          checkArgument(enforcementPercentage != null);
+          this.enforcementPercentage = enforcementPercentage;
+          return this;
+        }
+
+        /** The minimum amount of host for failure percentage ejection to be enabled. */
+        public Builder setMinimumHosts(Integer minimumHosts) {
+          checkArgument(minimumHosts != null);
+          this.minimumHosts = minimumHosts;
+          return this;
+        }
+
+        /**
+         * The request volume required for an address to be considered for failure percentage
+         * ejection.
+         */
+        public Builder setRequestVolume(Integer requestVolume) {
+          checkArgument(requestVolume != null);
+          this.requestVolume = requestVolume;
+          return this;
+        }
+
+        /** Builds a new instance of {@link FailurePercentageEjection}. */
+        public FailurePercentageEjection build() {
+          return new FailurePercentageEjection(threshold, enforcementPercentage, minimumHosts,
+              requestVolume);
+        }
+      }
+    }
+
+    /**
+     * Determine if outlier detection is at all enabled in this config.
+     */
+    boolean outlierDetectionEnabled() {
+      // One of the two supported algorithms needs to be configured.
```

Comment:
Outlier detection is enabled if one or both algorithms are configured, but the comment is misleading making it sound like an error if none or both are configured.  Also, the description of the return value belongs in the javadoc.

Reply:
I updated the comments.